### ISSUE TITLE
[env] mountain_car_continuous.rs: NaN/out-of-range force silently poisons state via step() and from_tensor()

### ISSUE THREAD

**Issue by anthonytorlucci** [anthonytorlucci/rlevo#774]
**File:** `crates/rlevo-environments/src/classic/mountain_car_continuous.rs`

## §1.1 — `step` does not reject non-finite/out-of-range force

`Environment::step` (currently around `mountain_car_continuous.rs:503-523`) reads
`action.force()` and passes it straight into `apply_physics` / the reward
computation with no finiteness check:

```rust
fn step(&mut self, action: MountainCarContinuousAction) -> Result<Self::SnapshotType, EnvironmentError> {
    let force = action.force();
    self.state = Self::apply_physics(self.state, force, &self.config);
    ...
```

`MountainCarContinuousAction::new` (line ~205) guards `is_finite() && (-1.0..=1.0).contains(&force)`,
but that invariant is bypassed by `from_slice`/`unchecked`/`from_tensor` (see below).
If `force = NaN` reaches `step`, `f32::clamp` propagates it, `vel`/`pos` become `NaN`,
`is_terminal` returns `false` (`NaN >= goal` is `false`), and
`reward = -0.1 * NaN * NaN = NaN` — the environment is now permanently poisoned with
no error surfaced. A large finite force (e.g. via `from_slice([1e30])`) similarly
produces `reward = -inf`.

Suggested fix: reject non-finite force in `step` with `EnvironmentError::InvalidAction`.

## §1.2 — `from_tensor` doesn't validate contents, only shape

`MountainCarContinuousAction::from_tensor` (currently `mountain_car_continuous.rs:573-588`)
validates only the tensor shape, then wraps the scalar with `Self::unchecked(v[0])`:

```rust
Ok(Self::unchecked(v[0]))
```

The core `TensorConvertible` doc contract requires `from_tensor` to reject invalid
*contents*, not just shape (cross-referencing `Action::is_valid`). Here `is_valid`
requires `is_finite() && abs() <= 1.0`, but `from_tensor` happily returns
`Ok(MountainCarContinuousAction(NaN))` or `Ok(...(5.0))`. This is the entry point
that feeds the §1.1 corruption path (e.g. from an RL agent's raw network output).

Suggested fix: route through `Self::new` so out-of-range/non-finite tensors error:

```rust
Self::new(v[0]).map_err(|e| TensorConversionError { message: e.to_string() })
```

## Verified still present (2026-07-24)

Checked against current source — both gaps are still live: `step` has no
finiteness check, and `from_tensor` still calls `Self::unchecked(v[0])` with no
validation.

Note: `§1.3` (raw-force reward being sound only when the pre-clip value is
already in bounds) is a downstream symptom of these two and needs no
independent fix — it becomes correct-by-construction once this issue is closed.

From the `crates/rlevo-environments/src/classic/mountain_car_continuous.rs` code
review (2026-07-01), items §1.1/§1.2/§1.3.

**Comment by anthonytorlucci** [comment]
Re-verified against the current tree by execution (not by reading the source), and this is still live. Two distinct mechanisms, which is worth keeping separate:

**NaN via `from_tensor`** — `MountainCarContinuousAction::from_tensor` (`mountain_car_continuous.rs:601-617`) calls `Self::unchecked(v[0])` with no check, unlike `Self::new` (`:210-218`). Feeding `NaN` through `from_tensor` → `step()` poisons the state **permanently**: `position: NaN, velocity: NaN, reward: NaN` on every subsequent step. Root cause is that `Bounds::clamp` (`rlevo-core/src/bounds.rs:184-186`) is `x.clamp(lo, hi)`, and `f32::clamp` propagates NaN — so the clamp is a silent no-op against NaN specifically, exactly as ADR 0065 already documents.

**Large finite out-of-range force** — different outcome, and the issue is right to treat it separately. With `force = 1e30` the state stays **finite** (normal physics, `position: -0.4388307`), but the reward is `-inf` every step, because `ctrl_cost` at `mountain_car_continuous.rs:536` squares the **raw, pre-clip** force rather than the physics-clamped value.

**Blast radius: this is live code, not unfinished.** `MountainCarContinuous` is wired into `bench/family.rs:50,79` via `RecordedEnvFamily`, which is the `Suite` factory surface `rlevo-benchmarks` drives. The `from_tensor` bypass is exactly the path a policy-network-driven agent adapter takes to turn raw tensor output into this env's action type.

**What #1115 / ADR 0078 changes here** — it does not fix this bug, but it changes the severity story. Because this env can emit a non-finite *reward* (not merely a poisoned state) and is reachable through the harness:

- *Before:* in a debug build the evaluator's `debug_assert!` panicked, and `catch_unwind` discarded every already-completed episode in that trial (aborting the suite under `fail_fast`). In release it silently poisoned `return/mean`.
- *Now:* the trial completes, the poisoned return is preserved rather than fabricated away, and `return/non_finite_steps` in the JSON `TrialReport` makes it diagnosable.

**Sibling defect, already tracked.** `PendulumAction::from_tensor` (`pendulum.rs:632-648`) has the identical `Self::unchecked(v[0])` shape — confirmed executed, NaN torque poisons Pendulum's state and reward the same way. That is already filed as #769, so nothing new needs filing.

One note for whoever fixes both: Pendulum's reward squares the **clamped** torque (`pendulum.rs:420`), so a large finite out-of-range torque does *not* poison Pendulum's reward the way it poisons this one (executed: reward stayed finite, ≈ -3.6 to -5.0, at `torque = 1e30`). The unchecked-construction defect is shared; the out-of-range-finite reward sub-case is **not**. A fix description claiming both envs are identically broken would be wrong.

Discrete `MountainCarAction` (`mountain_car.rs:595-611`) is a genuine near-miss needing no change — its `from_tensor` runs raw logits through `argmax` into a fixed discrete index, so there is no unchecked scalar to poison.
